table/block_based/block_based_table_builder.cc

```diff
@@ -1024,13 +1026,16 @@ void BlockBasedTableBuilder::Flush() {
                                              r->get_offset());
     r->pc_rep->EmitBlock(block_rep);
   } else {
-    WriteBlock(&r->data_block, &r->pending_handle, BlockType::kData);
+    CachableEntry<Block>* cached_entry = nullptr;
+    WriteBlock(&r->data_block, &r->pending_handle, BlockType::kData,
+               cached_entry);
   }
 }
 
-void BlockBasedTableBuilder::WriteBlock(BlockBuilder* block,
-                                        BlockHandle* handle,
-                                        BlockType block_type) {
+template <typename TBlocklike>
+void BlockBasedTableBuilder::WriteBlock(
```

Comment:
Making this a template could substantially increase the size of this hot code, which makes me worry about performance regression. Also I think we should try to make this a minimal fix for now for confidence in back-porting.

Reply:
Sure. Makes sense. I will try to fix it with other way.